### ISSUE TITLE
std::fmt::Pointer should be implemented for pointers to T: ?Sized

### ISSUE THREAD

**Issue by benw** [rust-lang/rust#30427]
Example:

```rust
fn main() {
    let x: Box<std::fmt::Debug> = Box::new("hello");
    println!("{:p}", x);
    // error: the trait `core::marker::Sized` is not implemented for the type `core::fmt::Debug`
}
```

I think Pointer should be impled as follows:

```
impl<T: ?Sized> Pointer for Box<T>
```

and likewise for borrowed refs, raw pointers, Arc etc.

**Comment by sfackler** [comment]
Boxed trait objects contain two pointers - one to the object and the other to the vtable. Should only one be printed?

**Comment by benw** [comment]
Oh, that is a very good point, thanks. This is not nearly as obvious as I thought. I was assuming that the data pointer would be printed, but if you see any possible downside to implementing that, then I'll be entirely satisfied with transmuting to TraitObject and printing the data field for myself.

**Comment by bluss** [comment]
Considering trait objects can be cast to think pointers, whatever pointer that yields is the one.

**Comment by durka** [comment]
This was fixed by #31479.
